confidential-assets/src/confidentialKeyRotation.ts

```diff
@@ -55,21 +57,31 @@ export class ConfidentialKeyRotation {
     this.currEncryptedBalance = args.currEncryptedBalance;
     this.currConfidentialAmount = args.currConfidentialAmount;
     this.newConfidentialAmount = args.newConfidentialAmount;
+
+    this.newEncryptedBalance = this.newConfidentialAmount.getAmountEncrypted(
+      this.newDecryptionKey.publicKey(),
+      this.randomness,
+    );
   }
 
   static FIAT_SHAMIR_SIGMA_DST = "AptosConfidentialAsset/RotationProofFiatShamir";
 
   static async create(args: CreateConfidentialKeyRotationOpArgs) {
-    const randomness = args.randomness ?? ed25519GenListOfRandom(ConfidentialAmount.CHUNKS_COUNT);
+    const {
+      randomness = ed25519GenListOfRandom(ConfidentialAmount.CHUNKS_COUNT),
```

Comment:
What about the provided randomness, we're just not using it now right? In that case `CreateConfidentialKeyRotationOpArgs` shouldn't accept it as an arg at all.

Reply:
It is being used in the constructor.  Line 63.